src/schema/v2/requestLocation.ts

```diff
@@ -35,34 +39,43 @@ export const RequestLocationField: GraphQLFieldConfig<void, ResolverContext> = {
   description: "A requested location",
   args: {
     ip: {
-      type: new GraphQLNonNull(GraphQLString),
+      type: GraphQLString,
     },
   },
-  resolve: async (_root, args, { requestLocationLoader }) => {
+  resolve: async (_root, args, { requestLocationLoader, ipAddress }) => {
+    const ip = args.ip || ipAddress
```

Comment:
**thought:** any reason why you would want to prioritise the argument instead of the ip included in the request? I feel like it would make more sense to prioritize the latter

Reply:
My thought was that if an IP address is passed as an argument, it is expected to be used. In cases where the client wants to request the location of an IP address that is different from the one in the header.

Reply:
sounds good - LGTM then

Reply:
Yea agree, prioritize `args.ip` if present